app/src/main/java/org/mifos/mobile/repositories/ReviewLoanApplicationRepository.kt

```diff
@@ -0,0 +1,45 @@
+package org.mifos.mobile.repositories
+
+import kotlinx.coroutines.flow.Flow
+import okhttp3.ResponseBody
+import org.mifos.mobile.models.payload.LoansPayload
+import org.mifos.mobile.ui.enums.LoanState
+
+interface ReviewLoanApplicationRepository {
+
+    fun insertData(
+        loanState: LoanState,
+        loansPayload: LoansPayload,
+        loanName: String,
+        accountNo: String,
+    )
+
+    fun insertData(
+        loanState: LoanState,
+        loanId: Long,
+        loansPayload: LoansPayload,
+        loanName: String,
+        accountNo: String,
+    )
+
+    fun getLoanProduct() : String?
+
+    fun getLoanPurpose() : String?
+
+    fun getPrincipal() : Double?
+
+    fun getCurrency() : String?
+
+    fun getSubmissionDate() : String?
+
+    fun getDisbursementDate() : String?
+
+    fun getLoanName() : String
+
+    fun getAccountNo() : String
+
+    fun getLoanState() : LoanState
+
+    suspend fun submitLoan(): Flow<ResponseBody>
```

Comment:
we need only this in our repository , I dont think we need rest of the functions that you have defined above. Any module should follow SRP i.e Single Responsibility Principle

Reply:
Actually, I was doing the same but inside the submit loan function some variables were required which is the reason I moved all of them to the repository. 
 `override suspend fun submitLoan(): Flow<ResponseBody> {
        loansPayload.productName = null
        loansPayload.loanPurpose = null
        loansPayload.currency = null

        return flow {
            emit(
                if (loanState == LoanState.CREATE) {
                    dataManager.createLoansAccount(loansPayload)
                } else {
                    dataManager.updateLoanAccount(loanId, loansPayload)
                }
            )
        }
    }`

Reply:
Ok then I am passing the loansPayload, loanState and loanId as parameters to the submitLoan function